### ISSUE TITLE
3d ignite classification doesn't work on multiple GPU's

### ISSUE THREAD

**Issue by bartmch** [Project-MONAI/MONAI#767]
**Describe the bug**
I want to use all available GPU's on a classification task comparable to the [ignite classification example](https://github.com/Project-MONAI/MONAI/blob/master/examples/classification_3d_ignite/densenet_training_dict.py). When I change the trainer from 'trainer = create_supervised_trainer(net, opt, loss, device, False, prepare_batch=prepare_batch)' to 'trainer = create_supervised_trainer(net, opt, loss, None, False, prepare_batch=prepare_batch)' I get the following error:
![image](https://user-images.githubusercontent.com/10415569/87671556-97856880-c782-11ea-8a2f-2f69be084293.png)

The original code works fine on one GPU.

**To Reproduce**
Run the [ignite classification example](https://github.com/Project-MONAI/MONAI/blob/master/examples/classification_3d_ignite/densenet_training_dict.py) code and replace **device** with **None**

I'm using the latest 0.2.0 code

**Comment by Nic-Ma** [comment]
Hi @bartmch ,

1. The ignite classification example in MONAI is based on ignite v0.3.0, I checked the ignite API code and I think it can't support multiple gpus training as it doesn't have `DataParallel` logic in it.
2. And I think `device=None` means CPU in ignite instead of all GPUs: https://github.com/pytorch/ignite/blob/v0.3.0/ignite/engine/__init__.py, that's the root cause for your error.

So I think if you want to train with multiple GPUs and DataParallel, you can refer to:
```py
# define training components
device = torch.device("cuda" if args.use_gpu else "cpu")
network = UNet(dimensions=3, in_channels=1, out_channels=2).to(device)  # Note to move to GPU before optimizer
loss_function = DiceLoss(to_onehot_y=True, softmax=True)
optimizer = torch.optim.Adam(network.parameters(), args.learning_rate)
lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=5000, gamma=0.1)
network = torch.nn.DataParallel(network)  # Note to enable DataParallel

trainer = ignite.engine.Engine(...)
```

Thanks.
